clang-tools-extra/clang-tidy/cppcoreguidelines/ProTypeMemberInitCheck.cpp

```diff
@@ -422,6 +422,28 @@ static const char *getInitializer(QualType QT, bool UseAssignment) {
   }
 }
 
+static bool isStdArray(QualType QT) {
+  const auto *RT = QT->getAs<RecordType>();
+  if (!RT)
+    return false;
+  const auto *RD = RT->getDecl();
+  if (!RD)
+    return false;
+
+  const IdentifierInfo *II = RD->getIdentifier();
+  if (!II)
+    return false;
+
+  if (II->getName() == "array") {
+    const NamespaceDecl *NS = dyn_cast<NamespaceDecl>(RD->getDeclContext());
```

Comment:
```suggestion
    const auto *NS = dyn_cast<NamespaceDecl>(RD->getDeclContext());
```

Type is spelled explicitly in same statement.